app/controllers/HomePageController.scala

```diff
@@ -46,23 +46,29 @@ class HomePageController @Inject() (
       doesGoodsRecordExist = goodsRecords.exists(_.goodsItemRecords.nonEmpty)
     } yield {
       val downloadLinkMessagesKey = getDownloadLinkMessagesKey(downloadDataSummary, doesGoodsRecordExist)
-      downloadDataSummary match {
-        case Some(downloadDataSummary) if downloadDataSummary.status == FileReadyUnseen =>
-          Ok(view(downloadReady = true, downloadLinkMessagesKey))
-        case _                                                                          =>
-          Ok(view(downloadReady = false, downloadLinkMessagesKey))
-      }
+      Ok(view(downloadReady(downloadDataSummary), downloadLinkMessagesKey))
     }
   }
 
-  private def getDownloadLinkMessagesKey(opt: Option[DownloadDataSummary], doesGoodsRecordExist: Boolean): String =
+  private def downloadReady(downloadDataSummary: Option[Seq[DownloadDataSummary]]): Boolean =
+    downloadDataSummary
+      .flatMap(
+        _.collectFirst {
+          case summary if summary.status == FileReadyUnseen => true
+        }
+      )
+      .getOrElse(false)
+
+  private def getDownloadLinkMessagesKey(opt: Option[Seq[DownloadDataSummary]], doesGoodsRecordExist: Boolean): String =
     if (doesGoodsRecordExist) {
-      opt.map(_.status) match {
-        case Some(FileInProgress) | Some(FileReadyUnseen) | Some(FileReadySeen) =>
-          "homepage.downloadLinkText.filesRequested"
-        case _                                                                  =>
-          "homepage.downloadLinkText.noFilesRequested"
-      }
+      opt
+        .flatMap(
+          _.collectFirst {
```

Comment:
here shouldn't it be if there are no summaries in the list then we show the link - we don't need to only look at the first one

Reply:
Good shout, I forget that we had removed the RequestFiles status, so if there is any there that indicates we need "homepage.downloadLinkText.filesRequested"